src/sys/windows/selector.rs

```diff
@@ -515,7 +516,7 @@ impl SelectorInner {
         socket.set_sock_state(Some(sock));
         unsafe {
             self.add_socket_to_update_queue(socket);
-            self.update_sockets_events()?;
+            self.update_sockets_events_if_polling()?;
```

Comment:
Maybe I missed something, but I don't understand the need for calling self.update_sockets_events_if_polling()?; during register/reregister, why can't we just defer the "update_sockets_events" until the next select2 call?

Reply:
If one thread is polling and another is registering a new I/O source we want to make sure the polling thread wakes up for events on the new source. Otherwise the polling thread might block forever, which is happening in the `registry_behind_arc` (also see #1135 and https://dev.azure.com/tokio-rs/Tokio/_build/results?buildId=3275&view=logs&jobId=429a383f-e486-57ac-3982-9ac65641a32a&taskId=1f926ffb-0ec7-5f9b-ab05-76fbb04597a8&lineStart=93&lineEnd=94&colStart=1&colEnd=1)